src/respond-reject.c

```diff
@@ -90,20 +99,49 @@ TmEcode RespondRejectFunc(ThreadVars *tv, Packet *p, void *data, PacketQueue *pq
         return TM_ECODE_OK;
 }
 
+
+static TmEcode RespondRejectInit(ThreadVars *tv, const void *initdata, void **data)
+{
+    RespondRejectThread *rrt = SCMalloc(sizeof(RespondRejectThread));
+    if (unlikely(rrt == NULL))
+        return TM_ECODE_FAILED;
+
+    rrt->force_iface = 0;
+
+    /* Get the iface name */
+    const char *conf_val;
+    if ((ConfGetValue("reject-iface", &conf_val)) == 1) {
+        if (strlen(conf_val) >= IFNAMSIZ) {
+            SCLogError(SC_ERR_INVALID_VALUE, "Interface name too long");
+        } else {
+            strlcpy(rrt->devname, conf_val, IFNAMSIZ);
+            rrt->force_iface = 1;
+            SCLogInfo("Sending reject packets to interface '%s'", conf_val);
+        }
+    }
+
+    return TM_ECODE_OK;
```

Comment:
CC       respond-reject.o
respond-reject.c:123:12: warning: Potential leak of memory pointed to by 'rrt'
    return TM_ECODE_OK;
           ^~~~~~~~~~~
1 warning generated.

Reply:
rrt is not passed back to 'data'?

Which makes me wonder... is this tested?

Reply:
Not enough, sorry :(